main.tf

```
@@ -30,53 +30,45 @@ data "aws_iam_policy_document" "default" {
   }
 }
 
-resource "aws_security_group" "default" {
-  count       = module.this.enabled ? 1 : 0
-  name        = module.this.id
-  vpc_id      = var.vpc_id
-  description = "Bastion host security group"
-
-  tags = module.this.tags
+data "aws_ami" "default" {
+  most_recent = "true"
 
-  # Optional block; skipped if var.allowed_cidr_blocks is empty
-  dynamic "ingress" {
-    for_each = length(var.allowed_cidr_blocks) > 0 ? [1] : []
+  dynamic "filter" {
+    for_each = var.ami_filter
     content {
-      protocol    = "tcp"
-      from_port   = 22
-      to_port     = 22
-      cidr_blocks = var.allowed_cidr_blocks
-      description = "Allow SSH ingress traffic from trusted CIDR Blocks"
+      name   = filter.key
+      values = filter.value
     }
   }
 
-  # Optional block; skipped if var.ingress_security_groups is empty
-  dynamic "ingress" {
-    for_each = length(var.ingress_security_groups) > 0 ? [1] : []
-    content {
-      from_port       = 0
-      to_port         = 0
-      protocol        = -1
-      security_groups = var.ingress_security_groups
-      description     = "Allow ALL ingress traffic from trusted Security Groups"
-    }
-  }
+  owners = var.ami_owners
+}
 
-  # Optional block; skipped unless var.egress_allowed is set to true
-  dynamic "egress" {
-    for_each = var.egress_allowed ? [1] : []
-    content {
+module "sg" {
+  source  = "cloudposse/security-group/aws"
+  version = "0.1.4"
+
+  description = "Bastion host security group"
+  rules = [
```

```suggestion
  rules = var.security_group_rules
```

And delete the rest.

---

Updated